pymc/backends/mcbackend.py

```diff
@@ -106,16 +105,26 @@ def __init__(
             {sname: stats_dtypes[fname] for fname, sname, is_obj in sstats}
             for sstats in stats_bijection._stat_groups
         ]
+        if "tune" in stats_dtypes and self.sampler_vars:
+            # expose driver-owned warmup marker via the sampler-stats API.
+            self.sampler_vars[0].setdefault("tune", stats_dtypes["tune"])
 
         self._chain = chain
         self._point_fn = point_fn
         self._statsbj = stats_bijection
         super().__init__()
 
-    def record(self, draw: Mapping[str, np.ndarray], stats: Sequence[Mapping[str, Any]]):
+    def record(
+        self,
+        draw: Mapping[str, np.ndarray],
+        stats: Sequence[Mapping[str, Any]],
+        *,
+        tune: bool | None = None,
+    ):
         values = self._point_fn(draw)
         value_dict = dict(zip(self.varnames, values))
         stats_dict = self._statsbj.map(stats)
+        stats_dict["tune"] = bool(tune)
```

Comment:
If we call it `in_warmup` I can change [this line in McBackend](https://github.com/pymc-devs/mcbackend/blob/366ab2d316c53170fc0d5663b2b20c8d718a4eb6/mcbackend/core.py#L154) to look for it.

This has the advantage of not colliding with sampler-emitted `"tune"` stats which thereby become an optional, sampler-owned sampler-stat.